parser/main/astnodes/var/Variable.c

```diff
@@ -0,0 +1,66 @@
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <stdbool.h>
+
+#include "Variable.h"
+#include "SimpleVar.h"
+
+#include "ast/ast.h"
+
+#include "token/TokenList.h"
+#include "token/TokenKeys.h"
+#include "token/token.h"
+
+struct Variable* makeVariable(struct TokenList* tokens, bool debug) {
+
+	if(debug){
+		printf("Variable(...) from ");
+		list_print(tokens);
+	}
+	
+	if(list_size(tokens) < 1){ return NULL; }
+
+	struct Variable* res = malloc(sizeof(struct Variable));
+	res->simpleVar = NULL;
+
+	res->memberAccess = NULL;
+
+	struct TokenList* copy = list_copy(tokens);
+
+	res->simpleVar = makeSimpleVar(copy,debug);
+	if(res->simpleVar == NULL){
+		freeTokenListShallow(copy);
+		free(res);
+		return NULL;
+	}
+
+	if (list_size(copy) >= 2) {
```

Comment:
too much nested if statements. maybe invert branches and add a goto to before the end of the function

Reply:
fixed in efe915d4